aspnetcore/performance/response-compression.md

```diff
@@ -0,0 +1,176 @@
+
+[!code-csharp[Main](response-compression/sample/FullSample/Startup.cs?name=snippet2&highlight=6)]
+
+[!code-csharp[Main](response-compression/sample/FullSample/CustomCompressionProvider.cs?name=snippet1)]
+
+Submit a request to the sample application with the `Accept-Encoding: custom` header and observe the response headers. The `Vary` and `Content-Encoding` headers are present on the response. The response body (not shown) isn't compressed by the sample, as there is no compression implementation in the `CustomCompressionProvider` class of the sample. However, the sample shows where you would implement such a compression algorithm.
+
+![Fiddler window showing result of a request with the Accept-Encoding header and a value of custom. The Vary and Content-Encoding headers are added to the response.](response-compression/_static/request-custom-compression.png)
+
+### MIME types
+The middleware specifies a default set of MIME types for compression:
+* `text/plain`
+* `text/css`
+* `application/javascript`
+* `text/html`
+* `application/xml`
+* `text/xml`
+* `application/json`
+* `text/json`
+
+You can replace or append MIME types with the Response Compression Middleware options. Note that wildcard MIME types, such as `text/*` are not supported.
+
+The sample application adds a MIME type for `image/svg+xml` and will compress and serve the ASP.NET Core banner image (**banner.svg**). The new MIME type is added to the default MIME types with the [`.Concat()` (concatenate) method](https://msdn.microsoft.com/en-us/library/bb302894(v=vs.110).aspx).
+
+[!code-csharp[Main](response-compression/sample/FullSample/Startup.cs?name=snippet2&highlight=7)]
+
+### Compression with secure protocol
+Compressed responses over secure connections can be controlled with the `EnableForHttps` option; however, enabling compression over a secure connection is unsafe, not recommended, and disabled by default. For more information, see [CRIME: Information Leakage Attack against SSL/TLS](https://blog.qualys.com/ssllabs/2012/09/14/crime-information-leakage-attack-against-ssltls).
+
+## Middleware ordering
+The position of Response Compression Middleware relative to other middleware in the pipeline is important. Any terminal middleware placed before Response Compression Middleware will prevent it from compressing the response. For example, if you place Static File Middleware before the Response Compression Middleware, your static files cannot be compressed. If you place Static File Middleware after Response Compression Middleware, your static files can be compressed.
+
```

Comment:
I certainly can do that. I didn't do it only because I thought it was clear-ish from the text. We also didn't add a code example at the same point in the *Response Caching Middleware* doc, where the same situation exists. https://docs.microsoft.com/en-us/aspnet/core/performance/caching/middleware#configuration

Let me know if you want it -- I'm certainly cool with it if you think it's best.